contracts/seed/Seed.sol

```diff
@@ -166,33 +167,29 @@ contract Seed {
     }
 
     /**
-      * @dev                     Buy seed tokens.
+      * @dev                     Buy and lock seed tokens.
       * @param _seedAmount       The amount of seed tokens to buy.
     */
     function buy(uint256 _seedAmount) public isActive allowedToBuy {
-        seedRemainder = seedRemainder.sub(_seedAmount);
         //  fundingAmount is an amount of fundingTokens required to buy _seedAmount of SeedTokens
         uint256 fundingAmount = (_seedAmount.mul(price)).div(PCT_BASE);
+
         // Funding Token balance of this contract;
         uint256 fundingBalance = fundingToken.balanceOf(address(this));
-        //  alreadyLockedSeedTokens is an amount of already locked SeedTokens without fee
-        uint256 alreadyLockedSeedTokens = (fundingBalance.mul(PCT_BASE)).div(price);
+
         //  feeAmount is an amount of fee we are going to get in seedTokens
         uint256 feeAmount = (_seedAmount.mul(uint256(PPM))).mul(fee).div(PPM100);
-        //  lockedSeedFee is an amount of fee we already need to pay in seedTokens
-        uint256 lockedSeedFee = (alreadyLockedSeedTokens.mul(uint256(PPM))).mul(fee).div(PPM100);
 
         // total fundingAmount should not be greater than the hardCap
         require( fundingBalance.
                   add(fundingAmount).
                   add((feeAmount.mul(price)).div(PCT_BASE)) <= hardCap,
             "Seed: amount exceeds contract sale hardCap");
 
-        // We are calculating that we are not exceeding balance of seedTokens in this contract
-        // balanceToBe = amountOfSeedAlreadyLocked + amountOfSeedToLock + SeedFee
-        require( seedToken.balanceOf(address(this)) >= (alreadyLockedSeedTokens.
-                                                        add(_seedAmount).
-                                                        add(feeAmount)),
+        // the amount of seed tokens still to be distributed
+        seedRemainder = (seedRemainder.sub(_seedAmount)).sub(feeAmount);
+
+        require( seedToken.balanceOf(address(this)) >= seedRemainder,
```

Comment:
That's an incorrect require statement, `seedRemainder` is going to get lower and lower, and you are checking that `seedToken` balance of the smart contract is >= to the value that is going less and less.
Instead you should check if `seedRemainder` >= 0, but it is not possible to implement completely correctly, because `seedRemainder` is uint, so you'll need to check if `seedRemainder` > 0, and unfortunately, you wiill always have 1 seedToken remaining in the smart contract

Maybe there is some other way...

Reply:
okay, we are already checking for the ```funding balance``` to be less than equal to ```hardCap```. And we are already transferring ```seedToken``` amount which is calculated using ```hardCap```. So I don't think we need a ```require``` for this.
We have only one way to transfer ```seedToken``` from ```Seed``` contract.

Reply:
Well, in any case it would be good to have this require just in case... Require statement doesn't take a lot of gas, but in return provides an additional layer of safety... + We are checking that fundingToken is under the cap, cap may be higher then the actual amount of the seed tokens we are going to distribute.

Reply:
require is still incorrect, it checks nothing tbh, we need to check that user cannot claim more seed token then we actually have on the contract

Reply:
Okay, wait, it's an incorrect change. Pushing the correct change now.